src/vm/ast/parser/mod.rs

```diff
@@ -175,21 +188,37 @@ pub fn lex(input: &str) -> ParseResult<Vec<(LexItem, u32, u32)>> {
                         }?;
                         Ok(LexItem::LiteralValue(str_value.len(), value))
                     },
-                    TokenType::QualifiedContractPrincipalLiteral => {
+                    TokenType::FullyQualifiedContractIdentifierLiteral => {
                         let str_value = get_value_or_err(current_slice, captures)?;
                         let value = match PrincipalData::parse_qualified_contract_principal(&str_value) {
                             Ok(parsed) => Ok(Value::Principal(parsed)),
                             Err(_e) => Err(ParseError::new(ParseErrors::FailedParsingPrincipal(str_value.clone())))
                         }?;
                         Ok(LexItem::LiteralValue(str_value.len(), value))
                     },
-                    TokenType::UnexpandedContractNameLiteral => {
+                    TokenType::SugaredContractIdentifierLiteral => {
                         let str_value = get_value_or_err(current_slice, captures)?;
                         let value = match str_value[1..].to_string().try_into() {
```

Comment:
This looks suspicious to me.  Is it possible that `str_value[1..].to_string().try_into()` can result in `Ok(...)` even if the encoded data in `str_value` is an error message?  What stops this from happening?

Reply:
What do you mean by encoded data in `str_value` is an error message? The `str_value` can only be the lexed string data from the lex matcher. If no `value` field was set by the lex matching pass, `get_value_or_err` returns an err, which the `?` operator would handle.

Reply:
Ah, I misread.  Nevermind.

I do have an unrelated question though -- we're sure that `str_value` has length strictly greater than 0, right?  i.e. there's no way any of our regexes could match an empty string as a match group?

Reply:
None of our regexes match an empty string, though several of the regexes _don't_ specify a named match group. We could add an assertion that the `munch_index` increases on every iteration (since if an empty string matched, that and an infinite loop would be the outcome).